SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -27,58 +27,177 @@ This SWIP is part of a series of SWIPs (but can be implemented on its own). To s
-Currently, Swarm is implementing the chequebook contract (with a base currency of Ether) to allow nodes to receive payments without doing on-chain transactions. While the chequebook contract is beautiful in its simplicity, it is expected that users of Swarm might prefer a different way of compensation for their services provided, especially if they are already participating in a payment network (e.g. Lumino, Raiden or Lightning network). 
+Currently, Swarm is implementing the chequebook contract (with a base currency of Ether). While the chequebook contract is beautiful in its simplicity, it can be expected that users of Swarm might prefer a different way of compensation for their services provided, especially if they are already participating in a payment network (e.g. Lumino, Raiden or Lightning network). 
 
-Storage providers might want to be compensated with a different currency (e.g. an ERC20 token) or they might want to settle their payment using a different blockchain. Furthermore, a storage provider offering other paid services will find appealing not to be forced to support multiple payment systems, but being able to consolidate the payments received under a single technology.
+Furthermore, storage providers might want to be compensated with a different currency or they might want to settle their payment using a different blockchain altogether. 
 
 Finally, new users of Swarm could bootstrap its participation in a payment channel network by providing storage services with zero cost of entry, as described in [Generalised Swap Swear and Swindle games (Tron & Fischer, 2019).](https://www.sharelatex.com/read/yszmsdqyqbvc) 
 
-When it becomes possible for nodes to set their preference for a payment module, developers will be incentivized to implement payment modules on Swarm as it will be easy for users to choose to pay with this module. It will also seamlessly enable multicurrency support and foster interoperation across blockchains without forcing participants to be tied to a single Blockchain or settlement technology.
+When it becomes possible for nodes to set their preference for a payment module, developers will be incentivized to implement payment modules on Swarm as it will be easy for users to choose to pay with this module. It will also seamlessly enable multicurrency support and foster interoperation across blockchains without forcing participants to be tied to a single blockchain or settlement technology.
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
 
-At a high level a payment module is responsible for:
+### Generic payment module
```

Comment:
I think a payment module should also be responsible for setting the payment threshold, as the costs of sending payment via one payment module might be different than sending a payment via another module. Hence a different payment threshold might be required. 
What do you think?

Reply:
I disagree. The thresholds are defined in terms of honey, not money, and therefore should not be handled by modules. Also, I can't see how this could work efficiently, and I fear it would add even more complexity and currency juggling which I don't think is needed.

Reply:
Good point! I haven't considered the unit of accounting for the payment threshold. I will think about it and come back with the conclusion.

Reply:
If the payment treshhold is added as a parameter which is defined by the payment module, then we could have `ether` or any other on-chain payment as a viable option for a payment module. If not, on-chain payments will probably be always too expensive to be used in Swarm.

Reply:
I will raise this as a question for research @holisticode, as they payment / disconnect treshholds are right now defined as a hardcoded value in the Swarm source code, but this is in my opinion not a good idea as it will be impossible to ever to change these thresholds when we run Swarm on a live network. 
Same logic applies to why we need a msgToHoney and a honeyToMoney oracle...

Reply:
https://swarmresear.ch/t/adjusting-payment-thresholds/42

Reply:
Yes I definitely support the idea of an oracle-based payment threshold

Reply:
We should include it in a different SWIP and start the discussing it.